src/smessage.cpp

```diff
@@ -3645,7 +3645,7 @@ int SecureMsgEncrypt(SecureMessage &smsg, const std::string &addressFrom, const
 
     if (!HMAC_Init_ex(&ctx, &key_m[0], 32, EVP_sha256(), NULL)
```

Comment:
This is a pretty cool technique, I must admit.

Diffie Hellman:
shared_secret_ec_key = (public_key_receiver * temporary_private_key)

shared_secret = SHA512(shared_secret_ec_key) 

key_e is used to encrypted the message with AES256 CBC and then key_m is used to authenticate the message.
key_e = The first 33 bytes of shared_secret 
keym = the last 33 bytes of the shared_secret.

key_m is the salt for the HMAC.

So if you want to mess with the ciphertext, iv or destination, you also need to have knowledge of the public_receiver_key.